doc/ext-pake/proposed/spake2plus-proposal-draft.md

```diff
@@ -0,0 +1,66 @@
+Introduction
+------------
+
+This document proposes protocol flow and high-level API changes required in PSA Cryptography API 1.1 PAKE Extension, to support the SPAKE2+ protocol.
+
+SPAKE2+ Version
+---------------
+
+SPAKE2+, an Augmented PAKE Draft 02, 10 December 2020 is considered for proposal.
+
+Link : [https://tools.ietf.org/pdf/draft-bar-cfrg-spake2plus-02.pdf](https://tools.ietf.org/pdf/draft-bar-cfrg-spake2plus-02.pdf)
+
+Remarks
+
+*   SPAKE2+, an Augmented PAKE Draft 08, 5 May 2022 is the latest draft version. Link : [https://datatracker.ietf.org/doc/pdf/draft-bar-cfrg-spake2plus-08](https://datatracker.ietf.org/doc/pdf/draft-bar-cfrg-spake2plus-08)
+*   Shared Secret Key generation is not compatible between Draft 02 and 08.
+*   As most SPAKE2+ implementations e.g. Matter Specification Version 1.0 are based on Draft 02, this version is being considered for better interoperability.
+
+Expected PAKE API Flow :
+------------------------------------
+![](SPAKE2PLUS.svg)
+
+
+
+#### Out of band set up
+
+SPAKE2+ has an offline initialization step where both parties may share parties identities and additional data (the context). This step is considered out of scope but these shared information may be required for SPAKE2+ protocol execution.
+
+**Shared Information** : ProverID, VerifierID and Context.
```

Comment:
Also the SPAKE2+ ciphersuite needs to be common between the participants.

This also needs to be encoded in the API. The current PAKE ciphersuite can encode the PAKE algorithm, the primitive (e.g. ECC over sec256p1), and a hash. For SPAKE2+, we also need to encode a KDF, and maybe also a MAC. However:

* The specification currently only uses HKDF in the table of cipher suites - can we just assume that *KDF* = HDFK?
* Can we assume that *cA* and *cB* are *KcA* and *KcB* (and avoid the MAC), or do we have use cases for using this in applications that specify the use of a MAC for key confirmation?

Reply:
The SPAKE2+ draft only includes HKDF as option for KDF in the given ciphersuite (Latest draft also doesn't give option for any other KDF). Therefore no need to include the KDF field in ciphersuite and assume it to be HKDF. Recommended MACs are HMAC and CMAC, therefore we should add MAC field in the ciphersuite